File: src/my_project/easy_problems/from1to50/remove_element.py
from typing import List

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:

        temp = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[temp] = nums[i]
                temp += 1

        return temp




solution = Solution()
print(solution.removeElement([2,2,3,5,6],2))

File: src/my_project_typescript/easy_problems_typescript/from1to50/removeElement.ts
function removeElement(nums: number[], val: number): number {
  let temp: number = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== val) {
      nums[temp] = nums[i];
      temp += 1;
    }
  }
  return temp;
}


console.log(removeElement([0,1,2,2,3,0,4,2],2));